apps/01PigLatin.js

```diff
@@ -7,11 +7,87 @@ prompt.start();
 
 function pigLatin(word) {
 
-    // Your code here
+  // Concept Check
+  //
+  // Please explain why we are using String.prototype.toLowerCase() here.
+  // For more info:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+  //
+  // Your Answer: We are using String.prototype.toLowerCase() to neutralize the
+  // user input and make it easier for our functions to analyze.
+  //
+  word  = word.toLowerCase();
 
+  // Concept Check
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer: Since an array starts with an index of 0, when we set the
+  // vowelIndex to -1, we are setting the index to start before the start of
+  // the array.
+  //
+  var vowelIndex = -1; // Set it to assume there are no vowels
+
+  if ( ( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('a');
+  }
+
+  if ( ( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('e');
+  }
+
+  if ( ( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('i');
+  }
+
+  if ( ( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('o');
+  }
+
+  if ( ( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('u');
+  }
+
+  if ( ( word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('y');
+  }
+
+  // Concept Check
+  //
+  // Why are we using String.prototype.slice() here? What role does vowelIndex play?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+  //
+  // Your Answer: We are using String.prototype.slice() in order to parse
+  // the user input. The vowelIndex tells the slice function where to end in
+  // the firstPart variable and where to begin in the restWord variable.
```

Comment:
You might call this parsing, but to me that is a stretch.

The goal here is separate the word into two parts at the first vowel.